pymeasure/instruments/hp/hp4145x.py

```diff
@@ -0,0 +1,1116 @@
+                  f"on channel {channels[channel]}")
+    return value
+
+
+class Status(IntEnum):
+    DATA_READY = 0b00000001
+    SYNTAX_ERROR = 0b00000010
+    END_STATUS = 0b00000100
+    ILLEGAL_PROGRAM = 0b00001000
+    BUSY = 0b00010000
+    SELF_TEST_FAIL = 0b00100000
+    RQS = 0b01000000
+    EMERGENCY = 0b10000000
+    FIXTURE_LID_OPEN = 0b10000010
+    SMU_SHUT_DOWN = 0b10001000
+    POWER_FAILURE = 0b10100000
+
+
+class VS(Channel):
+    _voltage_name = "VS"
+    _disabled = False
+    _voltage = 0
+
+    _channel_functions = {"VAR1": 1, "VAR2": 2, "CONST": 3, "VAR1'": 4}
+    _channel_function = 3
+
+    manual_flush = False
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._voltage_name = f"VS{self.id}"
+
+    @property
+    def voltage(self):
+        """
+        Set the output voltage of one of the VS channels.
+
+        In 'USER_MODE' directly sets the output voltage.
+        In 'SYSTEM_MODE' only allowed with :attr:`channel_function` = 'CONST'.
+        """
+        raise LookupError("Property can not be read.")
+
+    @voltage.setter
+    def voltage(self, value):
+        value = strict_range(value, [0, 20])
+
+        self._disabled = False
+        if self.parent.mode == 'USER_MODE':
+            self.write("DS {ch}, %f" % value)
+        else:
+            # SYSTEM_MODE
+            if self._channel_function == self._channel_functions['CONST']:
+                self._voltage = value
+                if not self.manual_flush:
+                    self.flush_source_setup()
+            else:
+                raise RuntimeError("In SYSTEM_MODE only allowed with channel_function = 'CONST'")
+
+        self.check_set_errors()
+
+    @property
+    def voltage_name(self):
+        """
+        Control the voltage name for `SYSTEM_MODE`'s channel definition.
+
+        Defaults to VS[1..2]. Must be unique otherwise the instrument returns a 'PROGRAM_ERROR'.
+
+        Returns not the actual state of the instrument but the state of the object.
+        """
+        raise LookupError("Property can not be read.")
+
+    @voltage_name.setter
+    def voltage_name(self, value):
+        if len(value.strip()) > 6 or len(value.strip()) < 1:
+            raise ValueError("Name must be less than six and at least one characters")
+
+        self._voltage_name = value.upper()
+        self._disabled = False
+        if not self.manual_flush:
+            self.flush_channel_definition()
+
+    @property
+    def disabled(self):
+        """
+        Set the disabling of the respective channel.
+        """
+        raise LookupError("Property can not be read.")
+
+    @disabled.setter
+    def disabled(self, value):
+        if not value:
+            raise ValueError(
+                "Channel can only be disabled, it gets enabled implicitly with the other functions")
+        self._disabled = True
+        self.write("DE VS {ch}")
+        self.check_set_errors()
```

Comment:
In this case, I suggest to use a method instead, as it is an action:

~~~python
def disable(self)...
~~~